src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/JsonFormat/SearchResultPackagesConverter.cs

```diff
@@ -31,13 +31,16 @@ public override void Write(Utf8JsonWriter writer, IPackageSearchMetadata value,
             writer.WriteStartObject();
             writer.WriteString(JsonProperties.PackageId, value.Identity.Id);
 
-            if (_exactMatch)
+            if (value.Identity.Version is not null)
```

Comment:
When is version null?

Reply:
I can not think of a scenario where it is null. However, in our implementation of `PackageIdentity` we allow version to be null (https://github.com/NuGet/NuGet.Client/blob/309f2e302dff9e6ba8a2f48ddf28133d68ff7eae/src/NuGet.Core/NuGet.Packaging/Core/PackageIdentity.cs#L24). This is to make sure we absolutely don't get Null exceptions